src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1310,9 +1320,17 @@ def import_realm_token_to_cluster(self, cluster_fsid, realm_name, zonegroup_name
             raise
 
     def check_user_in_second_cluster(self, cluster_url, cluster_token, username,
-                                     replication_zone_name):
+                                     replication_zone_name, system_user_param,
+                                     realm_name):
         logger.info("Checking for user %s in the second cluster", username)
-        path = f'api/rgw/zone/get_user_list?zoneName={replication_zone_name}'
+        params = {
+            "zoneName": f"zoneName={replication_zone_name}",
+            "realmName": f"realmName={realm_name}",
+        }
+        if system_user_param in params:
+            path = f"api/rgw/zone/get_user_list?{params[system_user_param]}"
```

Comment:
just curious, why can't we sent something like below always. if the realm name is empty it'll just sent an empty string which is fine isn't it? That could probably eliminate the other logic as well?
```suggestion
            path = f"api/rgw/zone/get_user_list?zone_name={replication_zone_name}&realm_name={realm_name}"
```

Reply:
Here realm name and replication_zone_name both always have some value, neither of them can be empty string and we need to send only one as a param based on the condition.